main.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    size = self.size
+    while i < size
+      is_a?(Range) ? yield(min + i) : yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    idx = 0
+    my_each do |elem|
+      yield(elem, idx)
+      idx += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      selected = []
+      my_each { |ele| selected << ele if yield ele }
+    else
+      selected = {}
+      my_each { |key, value| selected[key] = value if yield(key, value) }
+    end
+    selected
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_all?(arg = nil, &prc)
+    return true if !block_given? && arg.nil? && include?(nil) == false && include?(false) == false
+    return false unless block_given? || !arg.nil?
+
+    if block_given?
+      my_each { |elem| return false if prc.call(elem) == false }
+    elsif arg.class == Regexp
+      my_each { |elem| return false if arg.match(elem).nil? }
+    elsif arg.class <= Numeric || arg.class <= String
+      my_each { |elem| return false if elem != arg }
+    else
+      my_each { |elem| return false if (elem.is_a? arg) == false }
+    end
+    true
+  end
+
+  def my_any?(arg = nil, &prc)
+    return true if !block_given? && arg.nil? && my_each { |elem| return true if elem == true } && empty? == false
+    return false unless block_given? || !arg.nil?
+
+    my_each do |ele|
+      return true if prc.call(ele) == true
+    end
+    false
+  end
```

Comment:
#### my_any?

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2, 3].my_any?(Integer) => true
[1, 'demo', false].my_any?(Integer) => true
['demo', false, nil].my_any?(Integer) => false
```
- Yours is giving an error because your my_any? is expecting a proc but in these examples no proc is given
![image](https://user-images.githubusercontent.com/52209067/77497243-80f18500-6e12-11ea-9cea-54d813e4cb3f.png)

- Original any?
![image](https://user-images.githubusercontent.com/52209067/77497302-9ebeea00-6e12-11ea-8b20-eb8023b862f2.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/d/)   => true.
```
- Yours is giving an error because your my_any? is expecting a proc but in this example no proc is given
![image](https://user-images.githubusercontent.com/52209067/77497492-112fca00-6e13-11ea-842e-7dc7cabdd037.png)

- Original any?
![image](https://user-images.githubusercontent.com/52209067/77497535-26a4f400-6e13-11ea-882c-e3f09dec51ee.png)

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
[1, 2, 3].my_any?(3)   => true
```
- Yours is giving an error because your my_any? is expecting a proc but in this example no proc is given
![image](https://user-images.githubusercontent.com/52209067/77497579-4c31fd80-6e13-11ea-86a5-463155e74d3b.png)

- Original
![image](https://user-images.githubusercontent.com/52209067/77497612-610e9100-6e13-11ea-8986-45abb24c41bc.png)